src/vm/contract.rs

```diff
@@ -327,7 +371,31 @@ impl Ord for WitnessPos {
     fn cmp(&self, other: &Self) -> Ordering {
         assert!(self.timestamp > 0);
         assert!(other.timestamp > 0);
-        self.timestamp.cmp(&other.timestamp)
+        const BLOCK_TIME: i64 = 10 /*min*/ * 60 /*secs*/;
+        match (self.layer1, other.layer1) {
+            (a, b) if a == b => self.height.cmp(&other.height),
+            (Layer1::Bitcoin, Layer1::Liquid)
+                if (self.timestamp - other.timestamp).abs() >= BLOCK_TIME =>
+            {
+                Ordering::Greater
+            }
+            (Layer1::Liquid, Layer1::Bitcoin)
+                if (other.timestamp - self.timestamp).abs() >= BLOCK_TIME =>
```

Comment:
According to https://github.com/RGB-WG/RFC/issues/13, the fallback to chain ordering rule should happen when "the absolute timestamp difference is less than 10 mins", is this what you meant?

```suggestion
                if (self.timestamp - other.timestamp).abs() < BLOCK_TIME =>
            {
                Ordering::Greater
            }
            (Layer1::Liquid, Layer1::Bitcoin)
                if (other.timestamp - self.timestamp).abs() < BLOCK_TIME =>
```

Reply:
Yes, thank you for spotting that!